### PR TITLE
Add configuration option for time zone to DateConverter

### LINKED ISSUE #1320: Graylog improperly handling timestamps
I have a Cisco FireSIGHT that is sending logs to Graylog. Graylog is parsing the messages improperly and seting adding 7 hours to the timestamp. Even if it is assuming UTC because it cannot determine the time zone how is it that its adding 7 hours when I live in PDT which is minus 7 hours from UTC?

See an example message below

<118>Jul 22 15:53:51 btsdefense SFIMS: [119:15:1] http_inspect: OVERSIZE REQUEST-URI DIRECTORY [Impact: Potentially Vulnerable] From "btsips1" at Wed Jul 22 15:53:50 2015 UTC [Classification: Pornography was Detected] [Priority: 2] {tcp} 10.107.5.116:60603->10.106.240.176:9004

The message clearly has the UTC timezone in it. I have other devices that call out timezone the same way and they get tagged properly.

I have tried playing around with extractors to see if they will work but so far nothing has been successful.

All of my timestamps are set to PDT. Including the server that Graylog is installed on as well as Graylog config files.

capture

### PR DESCRIPTION
**Pull request by joschi**
Fixes #1320

### CODE REVIEW

**Comment by joschi** [comment]
Should this also be back-ported to 1.1.5?

File: graylog2-server/src/main/java/org/graylog2/inputs/converters/DateConverter.java [hunk 1]
```diff
@@ -39,20 +48,37 @@ public DateConverter(Map<String, Object> config) throws ConfigurationException {
             throw new ConfigurationException("Missing config [date_format].");
         }
 
-        dateFormat = ((String) config.get("date_format")).trim();
+        this.dateFormat = ((String) config.get("date_format")).trim();
+        this.timeZone = buildTimeZone(config.get("time_zone"));
+    }
+
+    private static DateTimeZone buildTimeZone(Object timeZoneId) {
```
Comment (edmundoa) [inline graylog2-server/src/main/java/org/graylog2/inputs/converters/DateConverter.java]: This method is exactly the same as the one in `FlexibleDateConverter`. Would it make sense to extract the method somewhere else? So far we have been using `org.graylog2.plugin.Tools` for that.

Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/inputs/converters/DateConverter.java]: You're right. I've opted to extract the common functionality into an abstract base class.

**Comment by edmundoa** [comment]
LGTM :shipit:

**Comment by joschi** [comment]
Back-ported into 1.1.x in 3066f9a405b88bada461b9888b99a43452181a2d.

**Pull request merged**